launcher/src/main/java/org/apache/spark/launcher/AbstractAppHandle.java

```diff
@@ -99,8 +100,6 @@ boolean isDisposed() {
    */
   synchronized void dispose() {
     if (!isDisposed()) {
-      // Unregister first to make sure that the connection with the app has been really
-      // terminated.
       server.unregister(this);
       if (!getState().isFinal()) {
```

Comment:
do we need this `if`? `setState` would do nothing if state is final.

Reply:
It's not necessary, but it makes it clear that the code only wants to override the state in that case without having to read the code for `setState`.

(Or I could call `setState(LOST, false)` explicitly, I guess.)